packages/web-app-files/tests/unit/components/SideBar/Actions/FileActions.spec.js

```diff
@@ -0,0 +1,411 @@
+import Vuex from 'vuex'
+import DesignSystem from 'owncloud-design-system'
+import stubs from '../../../../../../../tests/unit/stubs/index.js'
+import { createLocalVue, mount } from '@vue/test-utils'
+import FileActions from '../../../../../src/components/SideBar/Actions/FileActions.vue'
+
+import GetTextPlugin from 'vue-gettext'
+
+const localVue = createLocalVue()
+localVue.use(DesignSystem)
+localVue.use(Vuex)
+localVue.use(GetTextPlugin, {
+  translations: 'does-not-matter.json',
+  silent: true
+})
+
+const filesPersonalRoute = { name: 'files-personal' }
+
+const meta = {
+  files: {
+    name: 'Files',
+    id: 'files',
+    icon: 'folder'
+  },
+  mediaviewer: {
+    name: 'Mediaviewer',
+    id: 'mediaviewer',
+    icon: 'image'
+  },
+  'draw-io': {
+    name: 'Draw.io',
+    id: 'draw-io',
+    icon: 'grid_on'
+  },
+  'markdown-editor': {
+    name: 'MarkdownEditor',
+    id: 'markdown-editor',
+    icon: 'text'
+  }
+}
+
+const routes = [
+  'files-personal',
+  'files-favorites',
+  'files-shared-with-others',
+  'files-shared-with-me',
+  'files-public-list'
+]
+
+const editors = [
+  {
+    app: 'draw-io',
+    extension: 'drawio',
+    handler: null,
+    icon: null,
+    newTab: true,
+    routeName: 'draw-io-edit',
+    routes
+  },
+  {
+    app: 'mediaviewer',
+    extension: 'png',
+    handler: null,
+    icon: null,
+    newTab: false,
+    routeName: 'mediaviewer-media',
+    routes
+  },
+  {
+    app: 'markdown-editor',
+    extension: 'md',
+    handler: null,
+    icon: null,
+    newTab: false,
+    routeName: 'markdown-editor',
+    routes
+  }
+]
+
+const sideBars = [
+  {
+    app: 'details-item',
+    enabled: jest.fn(),
+    icon: 'info_outline'
+  },
+  {
+    app: 'actions-item',
+    enabled: jest.fn(),
+    icon: 'info_outline'
+  }
+]
+
+const apps = {
+  customFileListIndicators: [],
+  file: {
+    edit: false,
+    path: ''
+  },
+  fileEditors: editors,
+  fileSideBars: sideBars,
+  newFileHandlers: editors,
+  meta
+}
+
+function getWrapper(route, { filename = 'testfile', extension, extraActions = {}, type = 'file' }) {
+  const mountStubs = { ...stubs, 'oc-button': false }
+  const mountOptions = {
+    localVue,
+    store: createStore({}, filename, extension, type),
+    stubs: mountStubs,
+    mocks: {
+      $route: route,
+      publicPage: () => false
+    },
+    computed: {
+      actions() {
+        let actions = getActions(extraActions)
+        if (type === 'folder') {
+          actions = actions.filter(item => item.icon !== 'download')
+        }
+        return actions
```

Comment:
extra logic in tests?

Reply:
download option is not available for folders, so just creating the object accordingly